File: NDPServer/src/main/java/com/slender/config/manager/RequestConfigManager.java
package com.slender.config.manager;

import org.springframework.http.HttpMethod;
import org.springframework.http.MediaType;

import java.util.HashMap;
import java.util.Optional;

public final class RequestConfigManager {
    public record RequestConfig(
        HttpMethod method,
        MediaType mediaType,
        boolean requireToken
    ){
        public Optional<HttpMethod> getMethod() {
            return Optional.ofNullable(method);
        }
        public Optional<MediaType> getBodyType() {
            return Optional.ofNullable(mediaType);
        }
    }

    private final static HashMap<String,RequestConfig> requestConfigs=new HashMap<>();

    public RequestConfigManager addURIConfig(String uri, HttpMethod method, MediaType mediaType,boolean requireToken){
        requestConfigs.put(uri, new RequestConfig(method, mediaType,requireToken));
        return this;
    }

    public RequestConfigManager addPOSTURIConfig(String uri, boolean requireToken){
        requestConfigs.put(uri, new RequestConfig(HttpMethod.POST, MediaType.APPLICATION_JSON,requireToken));
        return this;
    }

    public RequestConfigManager addGETURIConfig(String uri, boolean requireToken){
        requestConfigs.put(uri, new RequestConfig(HttpMethod.GET, MediaType.APPLICATION_JSON,requireToken));
        return this;
    }

    public RequestConfigManager addPUTURIConfig(String uri, boolean requireToken){
        requestConfigs.put(uri, new RequestConfig(HttpMethod.PUT, MediaType.APPLICATION_JSON,requireToken));
        return this;
    }

    public RequestConfigManager addDELETEURIConfig(String uri, boolean requireToken){
        requestConfigs.put(uri, new RequestConfig(HttpMethod.DELETE, null,requireToken));
        return this;
    }

    public Optional<RequestConfig> findRequestConfig(String uri){
        return Optional.ofNullable(requestConfigs.getOrDefault(uri,null));
    }
}

File: NDPServer/src/main/java/com/slender/config/manager/RequestConfigManager.kt
package com.slender.config.manager

import org.springframework.http.HttpMethod
import org.springframework.http.MediaType
import java.util.*

class RequestConfigManager {

    data class RequestConfig(
        val method: HttpMethod? = null,
        val mediaType: MediaType? = null,
        val requireToken: Boolean
    ) {
        fun getMethod(): Optional<HttpMethod> = Optional.ofNullable(method)
        fun getBodyType(): Optional<MediaType> = Optional.ofNullable(mediaType)
    }

    companion object {
        private val requestConfigs = HashMap<String, RequestConfig>()
    }

    fun addURIConfig(uri: String, method: HttpMethod?, mediaType: MediaType?, requireToken: Boolean): RequestConfigManager {
        requestConfigs[uri] = RequestConfig(method, mediaType, requireToken)
        return this
    }

    fun addPOSTURIConfig(uri: String, requireToken: Boolean): RequestConfigManager {
        requestConfigs[uri] = RequestConfig(HttpMethod.POST, MediaType.APPLICATION_JSON, requireToken)
        return this
    }

    fun addGETURIConfig(uri: String, requireToken: Boolean): RequestConfigManager {
        requestConfigs[uri] = RequestConfig(HttpMethod.GET, null, requireToken)
        return this
    }

    fun addPUTURIConfig(uri: String, requireToken: Boolean): RequestConfigManager {
        requestConfigs[uri] = RequestConfig(HttpMethod.PUT, MediaType.APPLICATION_JSON, requireToken)
        return this
    }

    fun addDELETEURIConfig(uri: String, requireToken: Boolean): RequestConfigManager {
        requestConfigs[uri] = RequestConfig(HttpMethod.DELETE, null, requireToken)
        return this
    }

    fun findRequestConfig(uri: String): Optional<RequestConfig> {
        return Optional.ofNullable(requestConfigs[uri])
    }
}